### PR TITLE
[REVIEW] Upgrade BlazingSQL to cudf 0.10

### PR DESCRIPTION
**Pull request by aucahuasi**

### CODE REVIEW

File: src/CalciteInterpreter.cpp [hunk 1]
```diff
@@ -267,29 +268,38 @@ project_plan_params parse_project_plan(blazing_frame& input, std::string query_p
...
+			const bool is_literal_col = is_literal(cleaned_expression);
+
+			if (is_literal_col) {
+				int index = i;
+				gdf_dtype col_type = infer_dtype_from_literal(cleaned_expression);
+				output_type_expressions[i] = col_type;
+				gdf_column_cpp output;
+
+				if(col_type == GDF_STRING_CATEGORY){
+					const std::string literal_expression = cleaned_expression.substr(1, cleaned_expression.size()-2);
```
Comment (felipeblazing) [inline src/CalciteInterpreter.cpp]: look at https://github.com/BlazingDB/blazingdb-ral/blob/develop/src/LogicalFilter.cpp#L153

That is a WAY more efficient way of doing this. This is if you are going to use this as an input, For output you have to make the column obviously. But if you are using this as an input you should do something like that.

File: src/ColumnManipulation.cu [hunk 2]
```diff
@@ -42,7 +42,7 @@ __global__ void gather_bits(
 			if((element_index + bit_index) < num_indices){
 				Index permute_index = indices[element_index + bit_index];
```
Comment (felipeblazing) [inline src/ColumnManipulation.cu]: Doesnt gather bits already exist in cudf?

File: src/ColumnManipulation.cu [hunk 3]
```diff
@@ -68,7 +68,8 @@ __global__ void gather_bits(
 }
 
 void materialize_valid_ptrs(gdf_column * input, gdf_column * output, gdf_column * row_indices){
-	if (output->valid != nullptr){
+    //if (output->valid != nullptr and input->valid != nullptr){
```
Comment (felipeblazing) [inline src/ColumnManipulation.cu]: i think this already exists in cudf have we checked?

File: src/GDFColumn.cu [hunk 4]
```diff
@@ -283,7 +283,7 @@ void gdf_column_cpp::create_gdf_column(gdf_dtype type, size_t num_values, void *
 
     gdf_column_view(this->column, (void *) data, valid_device, num_values, type);
     this->column->dtype_info.category = nullptr;
-    this->column->dtype_info.time_unit = TIME_UNIT_NONE; // TODO this should not be hardcoded
+    this->column->dtype_info.time_unit = (type == GDF_DATE64 || type == GDF_TIMESTAMP ? TIME_UNIT_ms : TIME_UNIT_NONE); // TODO this should not be hardcoded
```
Comment (felipeblazing) [inline src/GDFColumn.cu]: this is going to change very soon all the times will have their own data types, this is just a comment, nothing to do about this

File: src/ColumnManipulation.cu (hunk 2 above)
Comment (aocsa) [reply src/ColumnManipulation.cu]: He buscado algo parecido y no he encontrado algo asi que este expuesto para su uso.

File: src/ColumnManipulation.cu (hunk 3 above)
Comment (aocsa) [reply src/ColumnManipulation.cu]: Lo mismo que lo anterior.

File: src/CalciteInterpreter.cpp (hunk 1 above)
Comment (aucahuasi) [reply src/CalciteInterpreter.cpp]: Thanks men, as far we now @felipeblazing the literal fix is working fine if we want to make it faster we can consider a change for the future. In any case @rommelDB can you please check if we can adapt the solution that @felipeblazing is suggesting and how much effort will cost us?

File: src/ColumnManipulation.cu (hunk 2 above)
Comment (aucahuasi) [reply src/ColumnManipulation.cu]: @felipeblazing Yeah also @jeanp413 suggested we can use the cudf one but is only a guess from our part, we don't know if it has feature parity with that code and as @aocsa mentioned above it seems that's not the case.

File: src/ColumnManipulation.cu (hunk 3 above)
Comment (aucahuasi) [reply src/ColumnManipulation.cu]: Thanks @aocsa

**Pull request merged**